src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -75,59 +74,53 @@ internal static IEnumerable<Claim> MergeClaims(IEnumerable<Claim> claims, IEnume
-                {
-                    if (claims.Where(i => i.Type == claim.Type).FirstOrDefault() == null)
-                    {
-                        result.Add(claim);
-                    }
-                }
-                return result;
+                if (claims.Where(i => i.Type == claim.Type).FirstOrDefault() == null)
+                    result.Add(claim);
             }
-            else
-                return result.Concat(subjectClaims);
+
+            return result;
         }
 
         /// <summary>
-        /// Gets ValueType of the claim from it's Value.
+        /// Gets the value type of the claim.
         /// </summary>
-        /// <param name="value">Represents value of a claim.</param>
-        /// <returns>String representing claim's ValueType.</returns>
+        /// <param name="value"> The claim value.</param>
+        /// <returns> The value type of the claim.</returns>
         internal static string GetClaimValueTypeFromValue(object value)
         {
-            if (value.GetType().Name == typeof(String).Name)
-            {
-                return ClaimValueTypes.String;
-            }
-            if (value.GetType().Name == typeof(Boolean).Name)
-            {
-                return ClaimValueTypes.Boolean;
-            }
-            if (value.GetType().Name == typeof(Int32).Name)
-            {
-                return ClaimValueTypes.Integer32;
-            }
-            if (value.GetType().Name == typeof(Int64).Name)
-            {
-                return ClaimValueTypes.Integer64;
-            }
-            if (value.GetType().Name == typeof(Double).Name)
+            if (value != null)
             {
-                return ClaimValueTypes.Double;
-            }
-            if (value.GetType().Name == typeof(DateTime).Name)
-            {
-                return ClaimValueTypes.DateTime;
-            }
-            if (value.GetType().Name == typeof(List<>).Name)
-            {
-                foreach (var item in (IList)value)
+                if (value.GetType().Name == typeof(String).Name)
+                    return ClaimValueTypes.String;
+
+                if (value.GetType().Name == typeof(Boolean).Name)
+                    return ClaimValueTypes.Boolean;
+
+                if (value.GetType().Name == typeof(Int32).Name)
+                    return ClaimValueTypes.Integer32;
+
+                if (value.GetType().Name == typeof(Int64).Name)
+                    return ClaimValueTypes.Integer64;
+
+                if (value.GetType().Name == typeof(Double).Name)
+                    return ClaimValueTypes.Double;
+
+                if (value.GetType().Name == typeof(DateTime).Name)
+                    return ClaimValueTypes.DateTime;
+
+                if (value.GetType().Name == typeof(List<>).Name)
                 {
-                    return GetClaimValueTypeFromValue(item);
+                    foreach (var item in (IList)value)
```

Comment:
Wouldn't this return the value type for the first item in the list only?

If so, should this be a for each loop?

Reply:
Yes, it returns value type for first item only. 

I used `foreach` here to handle the case when there are objects of different types in the list. From the below comment in JwtTokenUtilities.CreateDictionaryFromClaims method, looks like we have not decided yet on how to handle this case.
`// What to do if the 'ClaimValueType' is not the same.`

I kept `foreach `   in this method to have the comment below make more sense for someone looking at the code.
`// What to do if the value type of objects in the list are different.`